Add OSRS Wiki shop data extraction scripts

AshleyThew opened the pull request:
Wanted to extract shop information, so created scripts to generate the json objects.


Note that this was generated by ai, but seems to have done a good job based on the existing scripts:

Introduces scripts for extracting shop properties and items from the OSRS Wiki, including handling tabber structures and shop info parsing. Updates config.py to add shop-related paths. Provides a unified update.py entry point for the shop data extraction workflow.

Final note, could you enable github issues on this fork?

AshleyThew committed:
Add OSRS Wiki shop data extraction scripts

Introduces scripts for extracting shop properties and items from the OSRS Wiki, including handling tabber structures and shop info parsing. Updates config.py to add shop-related paths. Provides a unified update.py entry point for the shop data extraction workflow.

DayV-git commented:
Issues have been opened.

From glancing over, looks like a bit of the info is incorrect, for example all the currencies are in coins, even though some (e.g. Gabooty store) use other currencies. Infinite stock is also being converted to "\u221e" and should be handled better. 

Duplicate item info can be removed, like members/price (price is null everywhere, can just use value from item data).

AshleyThew commented:
Sounds good thanks, was a quick script made during lunch so will spend a bit more time refining.

Also, does it need to generate docs separately? I did only need the jsons but for completeness sake

AshleyThew committed:
Improve shop data accuracy

Add type to shop items (items/unknown)
Add item caching for fast lookups
Improve store table detection
Improve currency extraction

Dablakbandit commented:
@DayV-git
Have implemented the changes requested

AshleyThew commented:
Commented wrong account lol, confirming

DayV-git commented on scripts/shops/shops_items.py:
```diff
@@ -0,0 +1,521 @@
+                    'shop_info': shop_info,
+                    'items': shop_items
+                }
+                logging.info(f"    Found {len(shop_items)} items and shop info: {shop_info}")
```
Logging all shop info is a bit spammy, at most just log the shop name

AshleyThew replied:
Made to debug

DayV-git commented on scripts/shops/shops_properties.py:
```diff
@@ -0,0 +1,185 @@
+        # Check if the page has an infobox shop template
+        if '{{Infobox Shop' in wikitext or '{{infobox shop' in wikitext:
+            # Use a simple counter as ID since most shops don't have explicit IDs
+            shop_id = str(hash(page_title) % 1000000)  # Generate a consistent ID from page title
```
Hash output seems different for different versions/python installs, and is also unused as far as I can see. Can just use regular numbering, if needed at all

AshleyThew replied:
Removed hashed id

DayV-git commented on scripts/shops/shops_items.py:
```diff
@@ -0,0 +1,521 @@
+    return params
+
+
+def item_id_lookup(name: str) -> int | None:
```
Syntax is only valid for Python 3.10 or higher. Requirements use 3.6, so change this to be compatible

AshleyThew replied:
Removed

DayV-git commented on scripts/shops/shops_items.py:
```diff
@@ -0,0 +1,521 @@
+    logging.info(f"    - {total_items} total items")
+    logging.info(f"    - {len(shops_by_item)} unique items sold across all shops")
+
+    # Export both structures
```
Are both files necessary?

AshleyThew replied:
More for quick lookup

DayV-git reviewed (commented):
Normally you wouldn't want to add json changes with regular code changes, but since they are new it's fine here.

DayV-git commented:
> Sounds good thanks, was a quick script made during lunch so will spend a bit more time refining.
> 
> Also, does it need to generate docs separately? I did only need the jsons but for completeness sake

For now we can leave it, as there would be no use for it. I only keep the docs version of data files updated for consistency

AshleyThew committed:
Update shop data keys to use shop names

Replaced numeric keys with shop names in shops-wiki-page-text-processed.json, shops-wiki-page-text.json, and shops-wiki-page-titles.json for improved readability and maintainability. Updated related scripts to accommodate the new key format.

DayV-git reviewed (commented):
Check out the warnings from logger. Can ignore the non-items but there are some actual issues, e.g. gold decorative armour from castle wars shop

AshleyThew committed:
Add item IDs for Castle Wars gold set and fungicide spray

Replaced 'unknown' shop entries with correct item IDs for Castle Wars gold decorative set and fungicide spray in shop data files. Updated item lookup logic to match items by prefix if exact name is not found.

AshleyThew commented:
Added a check for item prefix to hopefully resolve this

DayV-git committed:
Ensure item_id_lookup uses consistent case

DayV-git commented:
Looks good. There are still prices missing for non-coin-currency items as they can't be inferred from item data. But this cant be done with wiki text scraping so out of scope.
